core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -374,7 +375,12 @@ memories_instantiate(const WASMModule *module, WASMModuleInstance *module_inst,
     for (i = 0; i < module->import_memory_count; i++, import++, memory++) {
         uint32 num_bytes_per_page = import->u.memory.num_bytes_per_page;
         uint32 init_page_count = import->u.memory.init_page_count;
-        uint32 max_page_count = import->u.memory.max_page_count;
+        uint32 max_page_count =
+            max_memory_pages == 0
+                ? import->u.memory.max_page_count
```

Comment:
How about handling `max_memory_pages == 0` inside `wasm_runtime_get_overwritten_max_mem`, and here it just calls, `wasm_runtime_get_overwritten_max_mem`?